src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -201,9 +137,15 @@ private ConstantValue DefaultSyntaxValue
                 if (_lazyDefaultSyntaxValue == ConstantValue.Unset)
                 {
                     var diagnostics = DiagnosticBag.GetInstance();
-                    if (Interlocked.CompareExchange(ref _lazyDefaultSyntaxValue, MakeDefaultExpression(diagnostics, ParameterBinder), ConstantValue.Unset) == ConstantValue.Unset)
+                    var defaultSyntaxValue = MakeDefaultExpression(diagnostics, ParameterBinderOpt);
+
+                    lock (_lazyLock)
```

Comment:
>lock (_lazyLock) [](start = 20, length = 16)

It is not clear why original synchronization is in sufficient for local functions. Could you elaborate please?